src/ngraph/builder/reshape.cpp

```diff
@@ -258,58 +258,37 @@ namespace ngraph
 
 shared_ptr<Node> builder::opset1::flatten(const Output<Node>& value, int axis)
 {
-    if (value.get_partial_shape().is_static())
+    // First dimension of output tensor is the product of [d_0, ... d_{axis-1}] dimensions of
+    // input tensor. The last dimension is the product of the rest of input tensor dimensions:
+    // [d_{axis}, ..., d_n]
+    shared_ptr<Node> output_shape;
+    if (axis == 0)
```

Comment:
This change should be documented in the PR title/description.
Maybe a separate PR should be created as well.

Reply:
I've improved description. Probably it should be in separate PRs, but flatten builder change would have to wait for dynamic backend support merge to master.

Reply:
Flatten PR could be made on top of dynamic backend.
After dynamic backend merge it's diff will be cleaned.